etcdserver/api/v3rpc/maintenance.go

```diff
@@ -147,6 +153,20 @@ func (ms *maintenanceServer) Status(ctx context.Context, ar *pb.StatusRequest) (
 	return resp, nil
 }
 
+func (ms *maintenanceServer) TransferLeadership(ctx context.Context, tr *pb.TransferLeadershipRequest) (*pb.TransferLeadershipResponse, error) {
+	if ms.rg.ID() != ms.rg.Leader() {
+		return nil, togRPCError(etcdserver.ErrNotLeader)
+	}
+
+	plog.Noticef("starting to transfer leadership from %s to %s", ms.rg.ID(), types.ID(tr.Transferee))
+	if err := ms.lt.TransferLeader(ctx, uint64(ms.rg.Leader()), tr.Transferee); err != nil {
+		return nil, togRPCError(err)
```

Comment:
is there already a warning or error message if there's a failure to transfer? the 'starting' message should probably be matched with some completion message even if it fails